dev/com.ibm.ws.persistence/src/com/ibm/wsspi/persistence/internal/DatabaseStoreImpl.java

```diff
@@ -484,10 +484,27 @@ protected String createOrmFileContentsForBatch(String[] entityClassNames) {
         orm
                         .append(" </entity>").append(EOLN);
         if (Arrays.asList(entityClassNames).contains("com.ibm.jbatch.container.persistence.jpa.JobInstanceEntityV2")) {
-            orm            
+            orm
+                            .append(" <entity class=\"com.ibm.jbatch.container.persistence.jpa.JobInstanceEntityV2\">").append(EOLN)
+                            .append("  <table name=\"").append(tablePrefix).append("JOBINSTANCE\"/>").append(EOLN)
+                            .append(" </entity>").append(EOLN);
+        }
+        if (Arrays.asList(entityClassNames).contains("com.ibm.jbatch.container.persistence.jpa.JobInstanceEntityV3")) {
+            orm
                             .append(" <entity class=\"com.ibm.jbatch.container.persistence.jpa.JobInstanceEntityV2\">").append(EOLN)
                             .append("  <table name=\"").append(tablePrefix).append("JOBINSTANCE\"/>").append(EOLN)
+                            .append(" </entity>").append(EOLN)
+                            .append(" <entity class=\"com.ibm.jbatch.container.persistence.jpa.JobInstanceEntityV3\">").append(EOLN)
+                            .append("  <table name=\"").append(tablePrefix).append("JOBINSTANCE\"/>").append(EOLN)
+                            .append("  <attributes>").append(EOLN)
+                            .append("   <element-collection name=\"groupName\" target-class=\"java.lang.String\">").append(EOLN)
+                            .append("    <collection-table name=\"").append(tablePrefix).append("GROUPASSOCIATION\">").append(EOLN)
+                            .append("     <join-column name=\"FK_JOBINSTANCEID\"/>").append(EOLN)
+                            .append("    </collection-table>").append(EOLN)
+                            .append("   </element-collection>").append(EOLN)
+                            .append("  </attributes>").append(EOLN)
                             .append(" </entity>").append(EOLN);
```

Comment:
What about defining an index? For our other tables with FK relationships, we generate an index right?

Reply:
add an index after the join-column-name stanza ala the job parameter table